core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -194,7 +194,7 @@ abstract class RDD[T: ClassTag](
   @transient private var partitions_ : Array[Partition] = null
 
   /** An Option holding our checkpoint RDD, if we are checkpointed */
-  private def checkpointRDD: Option[RDD[T]] = checkpointData.flatMap(_.checkpointRDD)
+  private def checkpointRDD: Option[CheckpointRDD[T]] = checkpointData.flatMap(_.checkpointRDD)
```

Comment:
I disagree. It's bad practice to have super general types everywhere. If we want different kinds of `CheckpointRDD`s we should have a common `CheckpointRDD` abstract class and plug in arbitrary implementations.